src/OrleansRuntime/Catalog/GrainCreator.cs

```diff
@@ -60,25 +63,38 @@ public Grain CreateGrainInstance(Type grainType, IGrainIdentity identity)
         /// <param name="grainType"></param>
         /// <param name="identity">Identity for the new grain</param>
         /// <param name="stateType">If the grain is a stateful grain, the type of the state it persists.</param>
-        /// <param name="storageProvider">If the grain is a stateful grain, the storage provider used to persist the state.</param>
+        /// <param name="persistenceProvider">If the grain is a stateful grain, the provider used to persist the state.</param>
         /// <returns></returns>
-        public Grain CreateGrainInstance(Type grainType, IGrainIdentity identity, Type stateType,
-            IStorageProvider storageProvider)
+        public Grain CreateGrainInstance(Type grainType, IGrainIdentity identity, Type stateType, IPersistenceProvider persistenceProvider)
         {
             //Create a new instance of the grain
             var grain = CreateGrainInstance(grainType, identity);
 
             var statefulGrain = grain as IStatefulGrain;
+            var logViewGrain = grain as ILogViewGrain;
 
-            if (statefulGrain == null)
-                return grain;
+            if (statefulGrain != null)
+            {
+                var storage = new GrainStateStorageBridge(grainType.FullName, statefulGrain, (IStorageProvider) persistenceProvider);
```

Comment:
I have given it some though but I am reluctant to invest a lot of energy into some general mechanism right now, when it is unclear what the benefit is - the two mechanisms exist side by side and there is very little overlap.

Right now, we have two provider mechanisms for grain state management (StorageProvider, LogViewProvider) that each are designed for a certain bridge type (StorageBridge, LogViewAdaptor<V,E,S>). Each has its own configuration section and its own ProviderManager. Each has its own semantics and test suites. 

They are designed for different grain types (StorageProvider is for Grain<T>, LogViewProvider is for QueuedGrain<V,E>, LogViewGrain<V,E> and JournaledGrain<V>).

The only overlap right now is that a regular storage provider can also serve as a log view provider (adding some metadata to the grain state), which is helpful because it means users can leverage existing storage providers when doing event sourcing or geo replication.

While it is possible we want more mechanisms in the future (e.g. other types of storage bridges), I don't think it is possible to fully anticipate the details without actually knowing what that interface is (for example, what services the runtime needs to expose), so I am not sure it makes sense to support some general extensibility mechanism without knowing what it is supposed to be used for concretely.




I don't see a need for other types anytime soon (the event sourcing is subsumed by LogViewProvider).